ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/BlockAttestationsPool.java

```diff
@@ -0,0 +1,156 @@
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.artemis.statetransition;
+
+import static tech.pegasys.artemis.datastructures.util.AttestationUtil.getAttesterIndicesIntoCommittee;
+import static tech.pegasys.artemis.datastructures.util.AttestationUtil.setBitsForNewAttestation;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.primitives.UnsignedLong;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Queue;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.PriorityBlockingQueue;
+import java.util.concurrent.atomic.AtomicBoolean;
+import org.apache.tuweni.bytes.Bytes32;
+import tech.pegasys.artemis.datastructures.operations.Attestation;
+import tech.pegasys.artemis.storage.ChainStorage;
+import tech.pegasys.artemis.util.SSZTypes.Bitlist;
+import tech.pegasys.artemis.util.SSZTypes.SSZList;
+import tech.pegasys.artemis.util.config.Constants;
+
+public class BlockAttestationsPool {
+
+  // TODO: memory clean up code brought over from ChainStorageClient,
+  // requires to be updated to new data structures to make it work.
+  // Memory Cleaning
+  //  private final ConcurrentHashMap<Bytes32, UnsignedLong> dataRootToSlot = new
+  // ConcurrentHashMap<>();
+
+  @VisibleForTesting
+  final ConcurrentHashMap<Bytes32, Bitlist> unprocessedAttestationsBitlist =
+      new ConcurrentHashMap<>();
+
+  @VisibleForTesting
+  final ConcurrentHashMap<Bytes32, Bitlist> processedAttestationsBitlist =
+      new ConcurrentHashMap<>();
+
+  private final int QUEUE_INITIAL_CAPACITY = 1024;
+
+  @VisibleForTesting
+  final Queue<Attestation> aggregateAttesationsQueue =
+      new PriorityBlockingQueue<>(
+          QUEUE_INITIAL_CAPACITY, Comparator.comparing(a -> a.getData().getSlot()));
+
+  public void addUnprocessedAggregateAttestationToQueue(Attestation newAttestation) {
+
+    Bytes32 attestationDataHash = newAttestation.getData().hash_tree_root();
+    AtomicBoolean oldBitlistPresent = new AtomicBoolean(true);
+    final Bitlist oldBitlist =
+        unprocessedAttestationsBitlist.computeIfAbsent(
+            attestationDataHash,
+            (key) -> {
+              oldBitlistPresent.set(false);
+              return newAttestation.getAggregation_bits().copy();
+            });
+
+    if (oldBitlistPresent.get()) {
+      if (!setBitsForNewAttestation(oldBitlist, newAttestation)) {
+        return;
+      }
+    }
+
+    aggregateAttesationsQueue.add(newAttestation);
+
+    //    ChainStorage.add(attestationDataHash, newAttestation.getData().getSlot(), dataRootToSlot);
```

Comment:
(nit) Do we need this comment?

Reply:
its part of the memory cleaning, so I'd keep it there